cli/src/movement_util.rs

```diff
@@ -214,16 +214,14 @@ fn choose_commit<'a>(
     choices.push("q".to_string());
     drop(formatter);
 
-    let choice = ui.prompt_choice(
+    let index = ui.prompt_choice(
         "enter the index of the commit you want to target",
         &choices,
```

Comment:
Would it make sense for `prompt_choice` to print the index with the option so we don't have to do that everywhere?

Reply:
I think a better option is to add `ui.prompt_choice_with(.., |choice: &str| choice == ...)`. In `jj next`/`prev`, it's nice if the commit can be selected by short change ID, for example.